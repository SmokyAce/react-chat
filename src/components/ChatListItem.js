import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { withStyles } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Badge from '@material-ui/core/Badge';
import Avatar from './Avatar';

const styles = theme => ({
  activeItem: {
    backgroundColor: theme.palette.grey[200],
  },
  badge: {
    border: `2px solid ${
      theme.palette.type === 'light' ? theme.palette.grey[50] : theme.palette.grey[900]
    }`,
  },
});

const ChatListItem = ({
  classes, disabled, title, chatId, active, createdAt, unread,
}) => (
  <ListItem
    button
    component={Link}
    to={`/chat/${chatId}`}
    className={active ? classes.activeItem : ''}
    disabled={disabled}
  >
    {unread > 0 ? (
      <Badge classes={{ badge: classes.badge }} badgeContent={unread} color='secondary'>
        <Avatar colorFrom={chatId}>{title}</Avatar>
      </Badge>
    ) : (
      <Avatar colorFrom={chatId}>{title}</Avatar>
    )}
    <ListItemText primary={title} secondary={moment(createdAt).fromNow()} />
  </ListItem>
);

ChatListItem.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  disabled: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
  chatId: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
  createdAt: PropTypes.string.isRequired,
  unread: PropTypes.number.isRequired,
};

export default withStyles(styles)(ChatListItem);
