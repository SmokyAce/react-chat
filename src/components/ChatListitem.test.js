/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ChatListItem from './ChatListItem';

jest.mock('./Avatar', () => () => 'Avatar');
jest.mock('moment', () => () => ({ fromNow: () => '2 days ago' }));

const mockProps = {
  chatId: '1',
  title: 'test',
  active: false,
  disabled: false,
  createdAt: '2018-03-08T13:57:58.706Z',
};

describe('<ChatList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ChatListItem {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  /* eslint-disable react/jsx-indent */
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatListItem {...mockProps} />
              </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly disabled', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatListItem {...mockProps} disabled />
              </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly active', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatListItem {...mockProps} active />
              </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
