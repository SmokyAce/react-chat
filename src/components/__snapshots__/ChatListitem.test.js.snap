// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList /> renders correctly 1`] = `
<a
  className="MuiButtonBase-root-80 MuiListItem-root-68 MuiListItem-default-71 MuiListItem-gutters-76 MuiListItem-button-77"
  href="/chat/1"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  Avatar
  <div
    className="MuiListItemText-root-83"
  >
    <span
      className="MuiTypography-root-89 MuiTypography-subheading-96 MuiListItemText-primary-86"
    >
      test
    </span>
    <p
      className="MuiTypography-root-89 MuiTypography-body1-98 MuiTypography-colorTextSecondary-122 MuiListItemText-secondary-87"
    >
      2 days ago
    </p>
  </div>
  <span
    className="MuiTouchRipple-root-124"
  />
</a>
`;

exports[`<ChatList /> renders correctly active 1`] = `
<a
  className="MuiButtonBase-root-80 MuiListItem-root-68 MuiListItem-default-71 MuiListItem-gutters-76 MuiListItem-button-77 ChatListItem-activeItem-66"
  href="/chat/1"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  Avatar
  <div
    className="MuiListItemText-root-83"
  >
    <span
      className="MuiTypography-root-89 MuiTypography-subheading-96 MuiListItemText-primary-86"
    >
      test
    </span>
    <p
      className="MuiTypography-root-89 MuiTypography-body1-98 MuiTypography-colorTextSecondary-122 MuiListItemText-secondary-87"
    >
      2 days ago
    </p>
  </div>
  <span
    className="MuiTouchRipple-root-124"
  />
</a>
`;

exports[`<ChatList /> renders correctly disabled 1`] = `
<a
  className="MuiButtonBase-root-80 MuiButtonBase-disabled-81 MuiListItem-root-68 MuiListItem-default-71 MuiListItem-gutters-76 MuiListItem-disabled-74 MuiListItem-button-77"
  href="/chat/1"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="-1"
>
  Avatar
  <div
    className="MuiListItemText-root-83"
  >
    <span
      className="MuiTypography-root-89 MuiTypography-subheading-96 MuiListItemText-primary-86"
    >
      test
    </span>
    <p
      className="MuiTypography-root-89 MuiTypography-body1-98 MuiTypography-colorTextSecondary-122 MuiListItemText-secondary-87"
    >
      2 days ago
    </p>
  </div>
</a>
`;
